pkg/devicetwin/dtcontext/dtcontext_test.go

```diff
@@ -0,0 +1,451 @@
+				t.Errorf("ModulesHealth =%v, want = %v", got.ModulesHealth, test.want.ModulesHealth)
+			}
+			if !reflect.DeepEqual(got.ModulesContext, test.want.ModulesContext) {
+				t.Errorf("ModulesContext =%v, want = %v", got.ModulesContext, test.want.ModulesContext)
+			}
+			if !reflect.DeepEqual(got.DeviceList, test.want.DeviceList) {
+				t.Errorf("DeviceList = %v, want = %v", got.DeviceList, test.want.DeviceList)
+			}
+			if !reflect.DeepEqual(got.DeviceMutex, test.want.DeviceMutex) {
+				t.Errorf("DeviceMutex = %v, want = %v", got.DeviceMutex, test.want.DeviceMutex)
+			}
+			if !reflect.DeepEqual(got.Mutex, test.want.Mutex) {
+				t.Errorf("Mutex = %v, want = %v", got.Mutex, test.want.Mutex)
+			}
+			if !reflect.DeepEqual(got.State, test.want.State) {
+				t.Errorf("State = %v, want = %v", got.State, test.want.State)
+			}
+		})
+	}
+}
+
+//TestCommTo is function to test CommTo().
+func TestCommTo(t *testing.T) {
+	// Failure Case
+	commChan := make(map[string]chan interface{})
+	wantErr := "Not found chan to communicate"
+	content := "FailureCaseArgument"
+	dtContext := &DTContext{
+		CommChan: commChan,
+	}
+	t.Run("FailureCase", func(t *testing.T) {
+		if err := dtContext.CommTo("test", content); err.Error() != wantErr {
+			t.Errorf("Failed to Get Channel")
+		}
+	})
+	//Success Case
+	testinterface := make(chan interface{})
+	commChan["test"] = testinterface
+	content = "SuccessCaseArgument"
+	dtContext = &DTContext{
+		CommChan: commChan,
+	}
+	t.Run("SuccessCase", func(t *testing.T) {
+		SendDataOnChannel := func() {
+			if err := dtContext.CommTo("test", content); err != nil {
+				t.Errorf("Unable to send data on channel: %v", err)
+			}
+		}
+		go SendDataOnChannel()
+		returnValue := <-testinterface
+		if returnValue != content {
+			t.Errorf("Got %v on channel, Want % v on Channel", returnValue, content)
+		}
+	})
+}
+
+//TestHeartBeat is function to test HeartBeat().
+func TestHeartBeat(t *testing.T) {
+	tests := []struct {
+		name    string
+		dtc     *DTContext
+		dtmName string
+		content interface{}
+		wantErr bool
+	}{
+		{
+			name: "SuccessCase",
+			dtc: &DTContext{
+				ModulesHealth: &sync.Map{},
+			},
+			dtmName: "Success",
+			content: "ping",
+			wantErr: false,
+		},
+		{
+			name: "FailureCase",
+			dtc: &DTContext{
+				ModulesHealth: &sync.Map{},
+			},
+			dtmName: "Failure",
+			content: "stop",
+			wantErr: true,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			dtc := &DTContext{
+				ModulesHealth: test.dtc.ModulesHealth,
+			}
+			if err := dtc.HeartBeat(test.dtmName, test.content); (err != nil) != test.wantErr {
+				t.Errorf("DTContext.HeartBeat() error = %v, wantErr %v", err, test.wantErr)
+			}
+		})
+	}
+}
+
+//TestGetMutex is function to test GetMutex().
+func TestGetMutex(t *testing.T) {
+	dtc := &DTContext{
+		DeviceMutex: &sync.Map{},
+	}
+	var testmutex *sync.Mutex
+	dtc.DeviceMutex.Store("FailureDeviceID", "testvalue")
```

Comment:
can't the value be an empty string ? 
also can we name "FailureDeviceID" to maybe "deviceA" or "deviceB" ? the device id has nothing wrong about it here , only the mutex value is wrong.